internal/util-collection/src/main/scala/sbt/internal/util/INode.scala

```diff
@@ -32,27 +32,25 @@ abstract class EvaluateSettings[Scope] {
   private[this] def getStatic[T](key: ScopedKey[T]): INode[T] =
     static get key getOrElse sys.error("Illegal reference to key " + key)
 
-  private[this] val transform: Initialize ~> INode = new (Initialize ~> INode) {
-    def apply[T](i: Initialize[T]): INode[T] = i match {
-      case k: Keyed[s, T] @unchecked => single(getStatic(k.scopedKey), k.transform)
-      case a: Apply[k, T] @unchecked =>
-        new MixedNode[k, T](
-          a.alist.transform[Initialize, INode](a.inputs, transform),
-          a.f,
-          a.alist
-        )
-      case b: Bind[s, T] @unchecked           => new BindNode[s, T](transform(b.in), x => transform(b.f(x)))
-      case v: Value[T] @unchecked             => constant(v.value)
-      case v: ValidationCapture[T] @unchecked => strictConstant(v.key)
-      case t: TransformCapture                => strictConstant(t.f)
-      case o: Optional[s, T] @unchecked =>
-        o.a match {
-          case None    => constant(() => o.f(None))
-          case Some(i) => single[s, T](transform(i), x => o.f(Some(x)))
-        }
-      case x if x == StaticScopes =>
-        strictConstant(allScopes.asInstanceOf[T]) // can't convince scalac that StaticScopes => T == Set[Scope]
-    }
+  private[this] val transform: Initialize ~> INode = λ[Initialize ~> INode] {
+    case k: Keyed[s, A1$] @unchecked => single(getStatic(k.scopedKey), k.transform)
```

Comment:
What's this dollar sign mean here?

Reply:
`A1$` is the name of the generated type parameter of `~>`'s apply, which was `T` before.

Basically this is what I found works that allows use to use polymorphic lamda syntax _even_ in places where we need the type param. It's more on the risky side of things in this diff. Btw there are 2 cases in InputTask.scala where I couldn't find any way to use the syntax, so it uses old school anonymous inner classes.